### PR TITLE
Update configuration.smk

### PR DESCRIPTION
**Pull request by siebrenf**
added minus (-) to the list of accepted characters in samples.tsv

### CODE REVIEW

File: rules/configuration.smk [hunk 1]
```diff
@@ -30,15 +30,20 @@ assert sorted(config['fqext'])[0] == config['fqext1']
...
+    ("\n" + config["samples"] + " may only contain letters, numbers and " +
+    "underscores (_), periods (.), or minuses (-).\n")
+# samples.columns = samples.columns.str.replace('[^A-Za-z0-9_.\-]+', '', regex=True)
 for column in samples.columns:
     assert column[0:7] not in ["Unnamed", ''], \
         ("\nEncountered unnamed column in " + config["samples"] +
          ".\nColumn names: " + str(', '.join(samples.columns)) + '.\n')
 # sanitize table content
 samples = samples.applymap(lambda x: x.strip() if isinstance(x, str) else x)
 samples = samples.replace(' ', '_', regex=True)
```
Comment (Maarten-vd-Sande) [inline rules/configuration.smk]: Almost! Lets also not allow spaces in sample names!

File: rules/configuration.smk [hunk 2]
```diff
@@ -30,15 +30,18 @@ assert sorted(config['fqext'])[0] == config['fqext1']
...
+assert not any(samples.columns.str.contains('[^A-Za-z0-9_.\-]+', regex=True)), \
+    ("\n" + config["samples"] + " may only contain letters, numbers and " +
+    "underscores (_), periods (.), or minuses (-).\n")
 for column in samples.columns:
     assert column[0:7] not in ["Unnamed", ''], \
         ("\nEncountered unnamed column in " + config["samples"] +
          ".\nColumn names: " + str(', '.join(samples.columns)) + '.\n')
 # sanitize table content
 samples = samples.applymap(lambda x: x.strip() if isinstance(x, str) else x)
-samples = samples.replace(' ', '_', regex=True)
```
Comment (Maarten-vd-Sande) [inline rules/configuration.smk]: I removed this one @siebrenf, or does that break anything?

Comment (Maarten-vd-Sande) [reply rules/configuration.smk]: Apparantly it does :angel:

Comment (Maarten-vd-Sande) [reply rules/configuration.smk]: The problem is that not everything is string in the samples dataframe, does it hurt to change everything to string? There is a column timepoint, do those have to be nrs or would string be okay?

Comment (siebrenf) [reply rules/configuration.smk]: That is exactly how it should happen! I actually added numbers in the example to make sure this exact thing would be interpreted right

**Pull request merged**